xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -543,6 +544,27 @@ void DX::DeviceResources::ResizeBuffers()
     }
   }
 
+  if (m_swapChain)
```

Comment:
As I said early merge this block with block started at L522

Reply:
This code is confusing for me (its functionality). I prefer not to touch it.

Reply:
It seems that this part of the code has become confusing. It needs refactoring (create an aux function outside to destroy & re-create swapchain). Yesterday I tried and created side effects (bugs in full screeen swiching or 3D).

Now there is duplicate code and it seems stupid but it works OK.

Reply:
zomg. no need thanks
```cpp
if (m_swapChain)
{
	BOOL bFullcreen = 0;
	m_swapChain->GetFullscreenState(&bFullcreen, nullptr);
	if (!!bFullcreen)
	{
		windowed = false;
	}

	bool hdr_capable;
	// check if swapchain needs to be recreated
	m_swapChain->GetDesc1(&scDesc);
	DetectDisplayHdrCapable(hdr_capable, isHdrEnabled);

	if ((scDesc.Stereo == TRUE) != bHWStereoEnabled
	  || (Is10BitSwapchain() == false && isHdrEnabled))
	{
		// check fullscreen state and go to windowing if necessary
		if (!!bFullcreen)
		{
			m_swapChain->SetFullscreenState(false, nullptr); // mandatory before releasing swapchain	        
		}
		m_swapChain = nullptr;
		m_deferrContext->Flush();
		m_d3dContext->Flush();
	}
}
```

Reply:
Thanks also :)